service/src/main/java/tds/assessment/repositories/impl/AssessmentMapper.java

```diff
@@ -18,82 +18,128 @@
+
+        // RULE:  An Assessment always has at least one Segment.  Create a "default" Segment from the data contained in
+        // the row that represents the Assessment.  This will account for assessments that do not have any segments.  In
+        // this case, the default Segment's assessmentKey will be set to the same value as the Assessment's key (to
+        // indicate the Segment is owned by the Assessment).
+        List<Segment> segments = new ArrayList<>();
+        segments.add(getSegment(row));
+        assessment.setSegments(segments);
 
         return Optional.of(assessment);
     }
 
-    private void updateSegmentWithRowData(Segment segment, ResultRow row, boolean isAssessmentRow) {
-        String assessmentKey = isAssessmentRow ? row.getString("assessmentSegmentKey") : row.getString("assessmentKey");
-        int position = isAssessmentRow ? 1 : row.getInt("segmentPosition");
-
-        segment.setSegmentId(row.getString("assessmentSegmentId"));
+    /**
+     * Build a {@link tds.assessment.Segment} that represents the {@link tds.assessment.Assessment}'s selection
+     * algorithm.
+     *
+     * @param row The database record containing data for the {@link tds.assessment.Segment}
+     * @return A new {@link tds.assessment.Segment}
+     */
+    private Segment getSegment(ResultRow row) {
+        // RULE:  The Segment's assessmentKey field should never be null.  If we're creating the "default" Segment (a
+        // segment for an Assessment that doesn't have any Segment records in the database), the "assessmentKey" field
+        // will be null, so the Segment's assessmentKey should be set to the whatever the Assessment's key is.  This
+        // means for the "default" Segment, its key and assessmentKey fields will be the same value.
+        String assessmentKey = row.getString("assessmentKey") == null
+            ? row.getString("assessmentSegmentKey")
+            : row.getString("assessmentKey");
+
+        // RULE:  A Segment's position can never be less than 1 (because segment position is 1-based).  If the segment
+        // position is less than 1 (which would be the case when segmentPosition is null in the result from the query -
+        // getInt() will return 0), set it to one.
+        int position = row.getInt("segmentPosition") < 1
+            ? 1
+            : row.getInt("segmentPosition");
+
+        // Populate the segment instance with data.  Because we are building a Segment, the "assessmentSegmentKey"
+        // represents the key (unique identifier) for this Segment (and not the Assessment).
+        Segment segment = new Segment(row.getString("assessmentSegmentKey"),
+            Algorithm.fromType(row.getString("selectionAlgorithm")));
         segment.setAssessmentKey(assessmentKey);
-        segment.setSelectionAlgorithm(row.getAlgorithm("selectionalgorithm"));
+        segment.setSegmentId(row.getString("assessmentSegmentId"));
```

Comment:
These were moved around as part of the refactoring/debugging effort.  I was trying to diagnose an issue w/the column labels.  It was easiest for me to verify that everything was spelled correctly and being pulled in properly when the `segment.set[whatever]` were put in the same order as the columns returned by the query.